neuro_flow/parser.py

```diff
@@ -1331,6 +1333,7 @@ def select_action(
         ret = {k: v for k, v in dct.items() if k in STATEFUL_ACTION}
     elif kind == ast.ActionKind.LOCAL:
         ret = {k: v for k, v in dct.items() if k in LOCAL_ACTION}
+        ret = select_shells(ctor, node, dct)
```

Comment:
```suggestion
        ret = select_shells(ctor, node, ret)
```
Or it will ignore filtering of keys.